api/src/paths/xlsx/transform.ts

```diff
@@ -0,0 +1,260 @@
+  persistParseErrors(),
+  getTransformationSchema(),
+  getTransformationRules(),
+  transformXLSX(),
+  persistTransformationResults(),
+  sendResponse()
+];
+
+POST.apiDoc = {
+  description: 'Transforms an XLSX survey observation submission file into a Darwin Core Archive file',
+  tags: ['survey', 'observation', 'xlsx'],
+  security: [
+    {
+      Bearer: [SYSTEM_ROLE.SYSTEM_ADMIN, SYSTEM_ROLE.PROJECT_ADMIN]
+    }
+  ],
+  requestBody: {
+    description: 'Request body',
+    content: {
+      'application/json': {
+        schema: {
+          type: 'object',
+          required: ['occurrence_submission_id'],
+          properties: {
+            occurrence_submission_id: {
+              description: 'A survey occurrence submission ID',
+              type: 'number',
+              example: 1
+            }
+          }
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: 'Transform XLSX survey observation submission OK'
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+export function persistParseErrors(): RequestHandler {
+  return async (req, res, next) => {
+    const parseError = req['parseError'];
+
+    if (!parseError) {
+      // no errors to persist, skip to next step
+      return next();
+    }
+
+    // file is not parsable, don't continue to next step and return early
+    // TODO add new status for "Transformation Failed" and insert new status record?
+    return res.status(200).json({ status: 'failed' });
+  };
+}
+
+export function getTransformationSchema(): RequestHandler {
+  return async (req, res, next) => {
+    defaultLog.debug({ label: 'getTransformationSchema', message: 'xlsx transform' });
+
+    const connection = getDBConnection(req['keycloak_token']);
+
+    try {
+      await connection.open();
+
+      const templateMethodologySpeciesRecord = await getTemplateMethodologySpecies(
+        req.body.occurrence_submission_id,
+        connection
+      );
+
+      const transformationSchema = templateMethodologySpeciesRecord?.transform;
+
+      if (!transformationSchema) {
+        // TODO handle errors if no transformation schema is found
+        // no schema to validate the template, generate error
+        // const submissionStatusId = await insertSubmissionStatus(
+        //   req.body.occurrence_submission_id,
+        //   'System Error',
+        //   connection
+        // );
+        // await insertSubmissionMessage(
+        //   submissionStatusId,
+        //   'Error',
+        //   `Unable to fetch an appropriate transformation schema for your submission`,
+        //   'Missing Validation Schema',
+        //   connection
+        // );
+        // await connection.commit();
```

Comment:
this commented code being left here is intentional right?

Reply:
Only partially.  I am not 100% certain what the workflow should be if transformation fails, but validation passes, etc.